VideoLocker/src/main/java/org/edx/mobile/view/adapters/CourseSequentialAdapter.java

```diff
@@ -0,0 +1,124 @@
+import org.edx.mobile.model.IUnit;
+import org.edx.mobile.model.IVertical;
+import org.edx.mobile.view.custom.PinnedSectionListView;
+
+import java.util.List;
+
+/**
+ * Used for pinned behavior.
+ */
+public abstract  class CourseSequentialAdapter extends CourseBaseAdapter
+    implements PinnedSectionListView.PinnedSectionListAdapter, SectionIndexer {
+
+
+    public CourseSequentialAdapter(Context context) {
+        super(context);
+    }
+
+    /**
+     * component can be null.
+     * @IComponent component should be ICourse
+     */
+    public void setData(IComponent component){
+        if ( !(component instanceof ISequential ) )
+            return;//
+        this.rootComponent = component;
+        mData.clear();
+        this.sections = new SectionRow[0];
+        if ( rootComponent != null ) {
+            ISequential course = (ISequential)rootComponent;
+            int sectionsNumber = course.getVerticals().size();
+            sections = new SectionRow[sectionsNumber];
+
+            int sectionPosition = 0, listPosition = 0;
+            for (char i=0; i<sectionsNumber; i++) {
+                IVertical vertical = course.getVerticals().get(i);
+                SectionRow section = new SectionRow(SectionRow.SECTION, vertical );
+                section.sectionPosition = sectionPosition;
+                section.listPosition = listPosition++;
+                sections[sectionPosition] = section;
+                mData.add(section);
+
+                List<IUnit> units = vertical.getUnits();
+                for (int j=0;j<units.size();j++) {
+                    SectionRow item = new SectionRow(SectionRow.ITEM, units.get(j) );
+                    item.sectionPosition = sectionPosition;
+                    item.listPosition = listPosition++;
+                    mData.add(item);
+                }
+
+                sectionPosition++;
+            }
+        }
+        notifyDataSetChanged();
+    }
+
+    @Override
+    public View getView(int position, View convertView, ViewGroup parent) {
+
+        int type = getItemViewType(position);
+
+        if (convertView == null) {
+            switch (type) {
+                case SectionRow.ITEM:
+                    convertView = mInflater.inflate(R.layout.row_course_outline_list, null);
+
+                    break;
+                case SectionRow.SECTION:
+                    convertView = mInflater.inflate(R.layout.row_section_header, null);
+                    break;
+            }
+        }
+        final SectionRow row = this.getItem(position);
+
+        switch (type) {
+            case SectionRow.ITEM:
+                IUnit unit = (IUnit)row.component;
+                //TODO - we will move it using ViewHolder pattern
+                ImageView image = (ImageView) convertView.findViewById(R.id.vertical_type);
+                image.setVisibility(View.INVISIBLE);
+                if ( "video".equalsIgnoreCase(unit.getCategory())){
```

Comment:
might be worth having a "categoryIsVideo" method somewhere, so we don't have to have the string around.